content/docs/user-guide/gems/reference/kythera-ai/spatial-query-system.md

```diff
@@ -0,0 +1,723 @@
+
+*   Returns the relative direction (forward, backward, left, right, up, or down) of the object as a KytVec3.
+    
+
+#### `DirectionTo(object)`
+
+*   Get a direction vector from the object to the candidate position.
+    
+
+#### `DirectionFrom(object)`
+
+*   Get a direction vector from the candidate position to the object.
+    
+
+#### **DotProduct(vec1, vec2)**
+
+*   Returns the dot product of the two vectors.
+    
+
+#### `Directness`
+
+*   Returns the directness of a given point: (`distanceFromQuerierToTarget` - `distanceFromPointToTarget`) / `distanceFromQuerierToPoint`)
+    
+
+#### **NormaliseAndClamp(function, min, max)**
+
+*   Normalises and clamps the float given by Function within the range of \[Min, Max\].
+    
+    *   Named parameters:
+        
+        *   function: float
+            
+        *   min: float
+            
+        *   max: float
+            
+
+#### `Absolute(function)`
+
+*   Returns the absolute value of the result from the input function. 
+    
+
+#### `Heading`
+
+*   Returns the cosine of the angle between the querier's forward direction and the direction to the current point.
+    
+
+#### `IsCoverOfType(coverType)`
+
+*   Returns true if the current point has the given cover type.
+    
+
+#### **NavRaycast(object, desiredResult)**
+
+*   Performs navmesh raycast from point to object. 
+    
+
+#### **OctreeRaycast(Object, DesiredResult, ShipRadius)**
+
+*   Performs octree raycast from point to Object.
+    
+
+#### **NavDistance(object, maxPathDistance)**
+
+*   Calculates the pathing distance between the current point and the given object. 
+    
+
+#### `OctreeSurfaceDistance(Object)`
+
+*   Calculates the approximate distance between the current point and the nearest octree surface
+    
+    *   Octree parameter "MaxRadius" needs to be high for this to be useful some distance from octree surfaces
+        
+    *   Object parameter should be Querier (it is only used to obtain the octree)
+        
+
+#### **GetBlackboard\[\*\]**
+
+*   Returns a value of specified type dependant on the function listed below, from the entity blackboard under the specified key.
+    
+    *   Named Parameters: 
+        
+        *    `Key (StringHash)`
+            
+    *   Available function types:
+        
+        *   _**GetBlackboardFloat**_(`Key`)
+            
+        *   _**GetBlackboardInt**(_`Key`)
+            
+        *   _**GetBlackboardBool**_(`Key`)
+            
+        *   _**GetBlackboardStringHash**_(`Key`)
+            
+        *   _**GetBlackboardPos**_(`Key`)
+            
+        *   _**GetBlackboardVec3**_(`Key`)
+
+## Debugging spatial queries
+
+### Debugging parser errors
+
+When working with parsed Spatial Queries (either from a .sqs file in Lumberyard, or an SQS Asset in Unreal Engine), you may sometimes make errors when writing your queries. This can lead to errors that cause your spatial query to be unparseable, meaning that Kythera will not be able to use it.
```

Comment:
Since this was written for O3DE, references to other engines are not necessary here.
```suggestion
When working with parsed Spatial Queries, you may sometimes make errors when writing your queries. This can lead to errors that cause your spatial query to be unparseable, meaning that Kythera will not be able to use it.
```